Models/AutoEncoder3D.py

```diff
@@ -1,87 +1,167 @@
+from abc import ABC
+
 import matplotlib.pyplot as plt
-from pytorch_lightning import LightningDataModule, LightningModule
 import numpy as np
+from torchinfo import summary
+import torchmetrics
+from monai.networks import blocks, nets
 import torch
-from collections import Counter
-import torchvision
-from torchvision import datasets, models, transforms
-from torchvision import transforms
+from torch import nn
 from pytorch_lightning.callbacks import ModelCheckpoint, EarlyStopping
-from pytorch_lightning import LightningDataModule, LightningModule, Trainer,seed_everything
-from torchsummary import summary
-import sys
+from pytorch_lightning import LightningDataModule, LightningModule, Trainer, seed_everything
+import sys, os
 import torchio as tio
-import sklearn
-from pytorch_lightning import loggers as pl_loggers
-import torchmetrics
-
+torch.cuda.empty_cache()
+torch.cuda.memory_summary(device=None, abbreviated=False)
 ## Module - Dataloaders
-from Dataloader.Dataloader import DataModule, DataGenerator, LoadSortDataLabel
+from DataGenerator.DataGenerator import DataModule, DataGenerator, QueryFromServer, SynchronizeData
+from Models.MixModel import MixModel
+
+## Main
+import toml
+from Utils.PredictionReports import PredictionReports
+from pathlib import Path
+
 
 ## Model
-class Classifier2D(LightningModule):
-    def __init__(self):
+class AutoEncoder3D(LightningModule, ABC):
+    def __init__(self, config):
         super().__init__()
         self.n_classes = 1
-        self.backbone = models.resnet50(pretrained=True) 
-        self.model= torch.nn.Sequential(
-            self.unet_model,
-            torch.nn.LazyLinear(128),
-            torch.nn.LazyLinear(self.n_classes)            
-        )
-        summary(self.model.to('cuda'), (2,160,160,40))
+        module_str = config['MODEL']['Backbone']
+        parameters = config['MODEL_PARAMETERS']
+        self.config = config
+
+        model_str = 'nets.' + module_str + '(**parameters)'
+        self.model = eval(model_str)
+        # only use network for features
+
+        summary(self.model.to('cuda'), (config['MODEL']['batch_size'], 1, *config['DATA']['dim']))
         self.accuracy = torchmetrics.AUC(reorder=True)
-        self.loss_fcn = torch.nn.BCEWithLogitsLoss()
+        self.loss_fcn = torch.nn.MSELoss()
 
-    def forward(self, x):
-        return self.model(x)
+    def forward(self, datadict):
+        key = list(datadict.keys())[0]
```

Comment:
this is not great, it ensure that you always use the first model_dict "[0]", but what if you want to do an autoencoder on Dose? Or what if someone define modality as Dose, CT, etc but wants to do it on CT?

Reply:
For the autoencoder, I want to use one modality each time, CT or Dose.  so the setting files just have one key.

Reply:
You may want to only use one modality, but some people might want to use the autoencoder weights loaded, so will need the autoencoder model. Just in general, it's bad practice to have hard coded variables.